include/miopen/miopen.h

```diff
@@ -337,8 +337,27 @@ typedef enum
     miopenBFloat16 = 5, /*!< 16-bit binary floating point (8-bit exponent, 7-bit fraction)
                            (Partially supported) */
     miopenDouble = 6,   /*!< 64-bit floating point (Partially supported) */
+    miopenHalfx4 = 7,   /*!<Pack of 4 16-bit floating point values. Suitable for vectorized tensor
+                           formats only.*/
+    miopenHalfx8 = 8,   /*!<Pack of 8 16-bit floating point values. Suitable for vectorized tensor
+                           formats only.*/
 } miopenDataType_t;
 
+/*! @ingroup tensor
+ * @enum miopenTensorLayout_t
+ * Tensor layouts supported by MIOpen.
+ * miopenTensorNCHW_VECT_C and miopenTensorCHWN_VECT_C layout are only supported with tensor data
+ * types miopenHalfx4 and miopenHalfx8. miopenTensorCHWN_VECT_C layout only support weight tensor.
+ */
+typedef enum
+{
+    miopenTensorNCHW        = 0, /*!< NCHW memory layout (Fully supported) */
+    miopenTensorNHWC        = 1, /*!< NHWC memory layout (Fully supported) */
+    miopenTensorCHWN        = 2, /*!< CHWN memory layout (Not supported) */
+    miopenTensorNCHW_VECT_C = 3, /*!< NCHW_VECT_C memory layout (Partially supported) */
+    miopenTensorCHWN_VECT_C = 4, /*!< CHWN_VECT_C memory layout (Partially supported) */
```

Comment:
[important] Unfortunately, API is cumbersome as @shurale-nkn mentions [here](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/1523#discussion_r860952292). 

(1) Do we foresee vectorization other than VECT_C? If not, then we can avoid adding a new API call and derive vectorized tensor layout internally, from the datatype.

(2) Otherwise we definitely need a new API call. But in this case I highly recommend developing an orthogonal API where pack (vector) size (x8, x4) and vectorized dimension (VECT_C) are separated from the datatype and tensor layout. Which means that the new API call would have two new parameters added (pack size and the index of dimension to be vectorized).

(3) The 3rd approach is adding a call like `miopenSetTensorDescriptorAttribute()` instead of this new `miopenSet4dTensorDescriptorWithLayout()`.

Of course, the above doesn't apply if maximum cudnn compatibility is paramount.

@junliume @JehandadKhan @pfultz2 What do you think?

Reply:
Thanks for the feedback @atamazov. I have posed similar questions to the frameworks team to get their input and will update here as soon as I hear back

Reply:
@JehandadKhan Great!

Reply:
@aska-0096  In place of the new members you created can we do something along the lines of 

```suggestion
    miopenTensorNCHWc4 = 3, /*!< NCHWc4 memory layout (Partially supported) */
    miopenTensorNCHWc8 = 4, /*!< NCHWc8 memory layout (Partially supported) */
    miopenTensorCHWNc4 = 4, /*!< CHWNc4 memory layout (Partially supported) */
    miopenTensorCHWNc8 = 4, /*!< CHWNc8 memory layout (Partially supported) */
```
cc/ @atamazov

Reply:
So the decision is to add new tensor layouts to the API. This seems acceptable.